### PR TITLE
`registerRootComponent` should accept FunctionalComponent

### PR DESCRIPTION
**Pull request by SimenB**
# Why

Allow passing `FunctionalComponent` to `registerRootComponent`, not just `ClassComponent`

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/21e68259f6fd43aba37e8112e3edb792975f8266/types/react/index.d.ts#L73

# How

Edited on GitHub

# Test Plan

No type error passing e.g. `const App: React.FC = () => <View />;`

### CODE REVIEW

File: packages/expo/src/launch/registerRootComponent.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ import withExpoRoot from './withExpoRoot';
 import { InitialProps } from './withExpoRoot.types';
 
 export default function registerRootComponent<P extends InitialProps>(
-  component: React.ComponentClass<P>
+  component: React.ComponentType<P>
 ): void {
   // @ts-ignore: TypeScript says ComponentClass<P> does not satisfy ComponentClass<any>
```
Comment (ide) [inline packages/expo/src/launch/registerRootComponent.tsx]: Please update these comments too, if the ts-ignore is still needed

File: packages/expo/src/launch/registerRootComponent.web.tsx [hunk 2]
```diff
@@ -5,7 +5,7 @@ import withExpoRoot from './withExpoRoot';
 import { InitialProps } from './withExpoRoot.types';
 
 export default function registerRootComponent<P extends InitialProps>(
-  component: React.ComponentClass<P>
+  component: React.ComponentType<P>
 ): void {
   const App = withExpoRoot(component);
   // @ts-ignore: TypeScript says ComponentClass<P> does not satisfy ComponentClass<any>
```
Comment (ide) [inline packages/expo/src/launch/registerRootComponent.web.tsx]: Same ehre

**Review by ide — CHANGES_REQUESTED** [review]

File: packages/expo/src/launch/registerRootComponent.tsx (hunk 1 above)
Comment (SimenB) [reply packages/expo/src/launch/registerRootComponent.tsx]: good call, it's not needed 🙂

File: packages/expo/src/launch/registerRootComponent.web.tsx [hunk 3]
```diff
@@ -5,11 +5,10 @@ import withExpoRoot from './withExpoRoot';
 import { InitialProps } from './withExpoRoot.types';
 
 export default function registerRootComponent<P extends InitialProps>(
-  component: React.ComponentClass<P>
+  component: React.ComponentType<P>
 ): void {
   const App = withExpoRoot(component);
-  // @ts-ignore: TypeScript says ComponentClass<P> does not satisfy ComponentClass<any>
-  const RootComponent = (...props) => <App exp={{}} {...props} />;
```
Comment (SimenB) [inline packages/expo/src/launch/registerRootComponent.web.tsx]: is this change correct?

It made `expo-module build` happy in this module.

Comment (SimenB) [reply packages/expo/src/launch/registerRootComponent.web.tsx]: I dunno how `AppRegistry.registerComponent` works, but if it expects to be passed a react component, it's correct 🙂

Comment (ide) [reply packages/expo/src/launch/registerRootComponent.web.tsx]: Yeah, I think this works and CI does type checks and passes.

**Pull request merged**